velox/exec/Exchange.h

```diff
@@ -159,18 +187,60 @@ class ExchangeQueue {
     return page;
   }
 
+  // Returns a guess for a typical reply unit size. An expected reply size is
+  // the requested bytes rounded up to the next multiple of the estimate. Actual
+  // reply sizes have no hard limit.
+  uint64_t expectedSerializedPageSize() const;
+
+  uint64_t requestableSpace() const;
+
+  // Marks that 'numRequests', each for 'bytes' have been started.
+  void recordRequestLocked(int32_t numRequests, uint64_t bytes) {
+    numPending_ += numRequests;
+    expectedBytes_ += bytes;
+  }
+
+  /// Records that a pending request arrived with 'bytes' of payload. Decreases
+  /// expected reply bytes.
+  void recordReplyLocked(int64_t bytes) {
```

Comment:
Is it expected that bytes here match bytes in recordRequestLocked?

Reply:
No, recordRequestLocked tried to record what we requested - which is a best effort guess on what the reply will be. Producer can produce an output of any size(but will try its stay within given limits).